tools/Raven.Debug/LogTrafficWatch/LogTrafficWatch.cs

```diff
@@ -0,0 +1,307 @@
+                    using (context.GetManagedBuffer(out var buffer))
+                    using (var parser = new UnmanagedJsonParser(context, state, "trafficwatch/receive"))
+                    using (var builder = new BlittableJsonDocumentBuilder(context, BlittableJsonDocumentBuilder.UsageMode.None, "readObject/singleResult", parser, state)) 
+                    // Write
+                    await using (var fileStream = new FileStream(file, FileMode.Append, FileAccess.Write, FileShare.Read, 32 * 1024, false))
+                    await using (var gZipStream = new GZipStream(fileStream, CompressionMode.Compress, false))
+                    using (var peepingTomStream = new PeepingTomStream(stream, context))
+                    using (var writer = new BlittableJsonTextWriter(context, gZipStream))
+                    {
+                        writer.WriteStartArray();
+                        var isFirst = true;
+
+                        while (fileStream.Length < maxFileSize)
+                        {
+                            if (_cancellationTokenSource.IsCancellationRequested)
+                            {
+                                writer.WriteEndArray();
+                                break;
+                            }
+
+                            try
+                            {
+                                var flushCount = 0;
+                                while (fileStream.Length < maxFileSize && _cancellationTokenSource.IsCancellationRequested == false)
+                                {
+                                    builder.Reset();
+                                    builder.Renew("trafficwatch/receive", BlittableJsonDocumentBuilder.UsageMode.None);
+
+                                    if (await UnmanagedJsonParserHelper.ReadAsync(peepingTomStream, parser, state, buffer).ConfigureAwait(false) == false)
+                                        continue;
+
+                                    await UnmanagedJsonParserHelper.ReadObjectAsync(builder, peepingTomStream, parser, buffer).ConfigureAwait(false);
+                                    using (var json = builder.CreateReader())
+                                    {
+                                        if (_changeTypes != null)
+                                        {
+                                            if (json.TryGet("Type", out TrafficWatchChangeType type) == false)
+                                                continue;
+                                            if (_changeTypes.Contains(type) == false)
+                                                    continue;
+                                        }
+
+                                        if (_database != null)
+                                        {
+                                            if(json.TryGet("DatabaseName", out LazyStringValue databaseName) == false ||
```

Comment:
can we use nameof?

Reply:
`TrafficWatchChangeBase` is internal and I don't want to change that.